### ISSUE TITLE
Suppress dotnet build summary

### ISSUE THREAD

**Issue by bgrainger** [Faithlife/FaithlifeBuild#10]
`dotnet build` normally prints a summary of the build output with all warnings and errors repeated. It looks something like:

```
  Determining projects to restore...
  Restored D:\Project.csproj (in 881 ms).
D:\File.cs(1,7): warning CS0618: 'Thing1' is obsolete: 'Use NewThing1.' [D:\Project.csproj]
D:\File.cs(2,7): warning CS0618: 'Thing2' is obsolete: 'Use NewThing2.' [D:\Project.csproj]
  Project -> D:\bin\Debug\Project.dll

Build succeeded.

D:\File.cs(1,7): warning CS0618: 'Thing1' is obsolete: 'Use NewThing1.' [D:\Project.csproj]
D:\File.cs(2,7): warning CS0618: 'Thing2' is obsolete: 'Use NewThing2.' [D:\Project.csproj]
    2 Warning(s)
    0 Error(s)
```

If a [problem matcher](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md) is installed, this logs duplicates of all the errors and warnings.

[According to Stack Overflow](https://stackoverflow.com/questions/44582185/duplicate-error-messages-in-dotnet-core/44780004#44780004), pass `/clp:NoSummary` on the `dotnet build` command line to suppress this summary.

Since the summary output is typically hidden deep within hundreds of lines of build output (from the targets following `build`), I think it makes sense to hide it by default (and maybe a `--show-summary` flag could be added to forcibly turn it back on via `/clp:Summary` if people need it?).

**Comment by bgrainger** [comment]
One downside of `-clp:NoSummary` is that it will suppress the following even on a successful build:

```
Build succeeded.
    0 Warning(s)
    0 Error(s)
```

This might typically be regarded as useful information to log (although it's typically redundant as the `build` target itself will show as passed or failed).

**Comment by ejball** [comment]
Can we fix the problem matcher? Surely there are other GitHub/dotnet users with this problem; they are all using this option?

**Comment by ejball** [comment]
No ETA: https://github.com/actions/runner/issues/504 https://github.com/actions/toolkit/issues/395

**Comment by ejball** [comment]
> Since the summary output is typically hidden deep within hundreds of lines of build output (from the targets following build)

If the build *fails*, which is why you'd want to see what failed, no more targets are run, and scrolling to the bottom of the output is a lovely way to see what went wrong.

**Comment by bgrainger** [comment]
> scrolling to the bottom of the output

is a waste of time, when a problem matcher will extract and display the errors in top-level GitHub Actions UI.

**Comment by ejball** [comment]
That's great, if every build anyone ever executes is in the context of a GitHub Action.

**Comment by bgrainger** [comment]
Most builds _probably_ are in the context of a different tool (GitHub Actions, Visual Studio, VSCode) that extract and show the errors through a User Interface.

I'm sure there are _some_ builds happening at the command line but a) this is actually an atypical scenario for us; b) they're generally sanity-checking when we expect the builds to pass. AFAIK, no one is actively developing via `.\build.cmd build`, parse CLI output by eye, fix errors in editor, GOTO 10.

**Comment by bgrainger** [comment]
> That's great

Sounds like we're agreed, then!

**Comment by ejball** [comment]
`--hide-summary` is easy to type once into a build script and call it good. But I concede.

Do you think it at all likely that anyone will ever *use* `--show-summary`? If not, it's just dead code and option clutter and we shouldn't add it.

**Comment by bgrainger** [comment]
> Do you think it at all likely that anyone will ever use --show-summary?

Not really, but it sounded like it might be important for your workflow? And while I don't want to create lots of dead code, I'm sympathetic to the idea of _not_ having overridable defaults in low-level tooling like this (so users have flexibility if they need it).

**Comment by ejball** [comment]
Could be an in-script option instead of a command-line option. If it was important to my workflow, I'd rather do that than type `--show-summary` every time I build.

**Comment by ejball** [comment]
Fixed in 5.6.0.
